### ISSUE TITLE
Display Archive Hierarchies Using Multi-Scope Search

### ISSUE THREAD

**Issue by roamye** [project-lux/lux-frontend#458]
**Problem Description:**
Archive hierarchies currently query for Collections and Objects separately. This prevents results from being combined, and their order may not always be correct.

**Expected Behavior/Solution:**
Archive hierarchies do a combined search for Sets and Objects using the multi scope ( new HAL link `lux:objectOrSetMemberOfSet`)

**Requirements:**
- archive hierarchies display combined results for Collections and Objects
- Archive hierarchies display results in the correct order

**Needed for promotion:**
If an item on the list is not needed, it should be crossed off but not removed.

- [ ] Wireframe/Mockup - Mike
- [ ] Committee discussions - Sarah
- [ ] Feasibility/Team discussion - Sarah
- [ ] Backend requirements - TBD
- [ ] Frontend requirements- TBD
- [ ] Are new regression tests required for QA - Amy
- [ ] Questions
- List of questions for discussions. Answers should be documented within the issue.

**UAT/LUX Examples:**

Before change (PRD) - https://lux.collections.yale.edu/view/set/0620b4d2-588a-4a68-8718-fa01059d57be
1. Expand the CONNECTICUT list
2. The list displays Sets first, then Objects. E.g. the beginning of the list is in this order:
  - Bridgeport
  - Hartford
  - New Haven
  - Stamford

After change (TST) - https://lux-front-tst.collections.yale.edu/view/set/0620b4d2-588a-4a68-8718-fa01059d57be
1. Expand CONNECTICUT list
2. The list displays the list in sort order, E.g. the beginning of the list is in this order:
  - Bridgeport
  - Essex
  - Greenwich
  - Hartford 

**Dependencies/Blocks:**
- Blocked By:
  - https://github.com/project-lux/lux-marklogic/issues/318
  - https://github.com/project-lux/lux-middletier/issues/135
- Blocking: #367 

**Related Github Issues:**
N/A

**Related links:**
Bugherd: https://www.bugherd.com/projects/284041/tasks/2548

**Wireframe/Mockup:**
Existing:

![Image](https://github.com/user-attachments/assets/5e8db6f8-303b-44e6-b3a5-f1c0dac85e5a)

Expected: 

![Image](https://github.com/user-attachments/assets/6a190385-e588-4c3e-8864-5cf4fbe955f9)

**Comment by roamye** [comment]
@clarkepeterf - what is a good uat example for this issue?

**Comment by roamye** [comment]
Note: https://github.com/project-lux/lux-marklogic/issues/318 was closed as part of the 11-18 milestone. It is unreviewable and must be reviewed within this issue once it is complete.

**Comment by clarkepeterf** [comment]
Middle tier will create a HAL link for `lux:objectOrSetMemberOfSet`

**Comment by clarkepeterf** [comment]
And the response will have Objects and Collections combined in the results, like this:

```
{
    "@context": "https://linked.art/ns/v1/search.json",
    "id": "https://lux.collections.yale.edu/api/search/multi?q=%7B%22OR%22%3A%5B%7B%22memberOf%22%3A%7B%22id%22%3A%22https%3A%2F%2Flux.collections.yale.edu%2Fdata%2Fset%2F87c14712-9239-4176-a112-913ff797481b%22%7D%7D%2C%7B%22memberOf%22%3A%7B%22id%22%3A%22https%3A%2F%2Flux.collections.yale.edu%2Fdata%2Fset%2F87c14712-9239-4176-a112-913ff797481b%22%7D%7D%5D%7D&page=1&pageLength=20&sort=archiveSortId",
    "type": "OrderedCollectionPage",
    "partOf": [
        {
            "id": "https://lux.collections.yale.edu/api/search-estimate/multi?q=%7B%22OR%22%3A%5B%7B%22memberOf%22%3A%7B%22id%22%3A%22https%3A%2F%2Flux.collections.yale.edu%2Fdata%2Fset%2F87c14712-9239-4176-a112-913ff797481b%22%7D%7D%2C%7B%22memberOf%22%3A%7B%22id%22%3A%22https%3A%2F%2Flux.collections.yale.edu%2Fdata%2Fset%2F87c14712-9239-4176-a112-913ff797481b%22%7D%7D%5D%7D",
            "type": "OrderedCollection",
            "label": {
                "en": [
                    "Multiple Types"
                ]
            },
            "summary": {
                "en": [
                    "Records representing multiple types that match your search."
                ]
            },
            "totalItems": 49
        }
    ],
    "orderedItems": [
        {
            "id": "https://lux.collections.yale.edu/data/set/4c0a5c89-c9dd-4779-ad5b-fd8098ee3c8f",
            "type": "Set"
        },
        {
            "id": "https://lux.collections.yale.edu/data/set/ca21e297-aa1e-439e-a5b8-3903682e3e60",
            "type": "Set"
        },
        {
            "id": "https://lux.collections.yale.edu/data/set/315cec4a-74bd-49f9-91f2-d5449c5b3f57",
            "type": "Set"
        },
        {
            "id": "https://lux.collections.yale.edu/data/set/eaae9b56-fd5f-4211-9112-019ea3bc4151",
            "type": "Set"
        },
        {
            "id": "https://lux.collections.yale.edu/data/set/0620b4d2-588a-4a68-8718-fa01059d57be",
            "type": "Set"
        },
        {
            "id": "https://lux.collections.yale.edu/data/set/d07aaf36-9ea4-422d-820b-a4fb96745e41",
            "type": "Set"
        },
        {
            "id": "https://lux.collections.yale.edu/data/digital/c208e695-9330-4457-a59e-02eca360f756",
            "type": "DigitalObject"
        },
        {
            "id": "https://lux.collections.yale.edu/data/digital/deac225b-eb3d-4d71-85e0-cfa904231f34",
            "type": "DigitalObject"
        },
        {
            "id": "https://lux.collections.yale.edu/data/set/e58ae36f-1ef5-41ab-a36c-5abb5d063f5e",
            "type": "Set"
        },
        {
            "id": "https://lux.collections.yale.edu/data/set/9f01d2cd-8dd1-4d85-82ca-52dfc9c86a09",
            "type": "Set"
        },
        {
            "id": "https://lux.collections.yale.edu/data/digital/c808d3d8-2be1-4e4f-8a5d-621f0bf263b2",
            "type": "DigitalObject"
        },
        {
            "id": "https://lux.collections.yale.edu/data/set/72841d45-a9fe-44bc-98f7-5ba41490078e",
            "type": "Set"
        },
        {
            "id": "https://lux.collections.yale.edu/data/digital/0d9ccda4-85c7-404a-a314-1516dec5141f",
            "type": "DigitalObject"
        },
        {
            "id": "https://lux.collections.yale.edu/data/set/2e28ff53-4703-47b0-84a1-6ee20dea72d9",
            "type": "Set"
        },
        {
            "id": "https://lux.collections.yale.edu/data/set/f940f526-7a3c-4041-a188-c6a6078b6ae3",
            "type": "Set"
        },
        {
            "id": "https://lux.collections.yale.edu/data/set/de6dd931-4a53-4b06-a4d8-567174f302a2",
            "type": "Set"
        },
        {
            "id": "https://lux.collections.yale.edu/data/set/a2ddb232-925e-4471-8db9-d7cd0c698645",
            "type": "Set"
        },
        {
            "id": "https://lux.collections.yale.edu/data/set/3cecba28-7353-44af-abe7-81be73b3e043",
            "type": "Set"
        },
        {
            "id": "https://lux.collections.yale.edu/data/set/993ab423-54d2-4f77-9cc5-2bb8f226e2d7",
            "type": "Set"
        },
        {
            "id": "https://lux.collections.yale.edu/data/set/188f86fe-ed1d-4a3d-b382-c7f409f1d6d7",
            "type": "Set"
        }
    ],
    "next": {
        "id": "https://lux.collections.yale.edu/api/search/multi?q=%7B%22OR%22%3A%5B%7B%22memberOf%22%3A%7B%22id%22%3A%22https%3A%2F%2Flux.collections.yale.edu%2Fdata%2Fset%2F87c14712-9239-4176-a112-913ff797481b%22%7D%7D%2C%7B%22memberOf%22%3A%7B%22id%22%3A%22https%3A%2F%2Flux.collections.yale.edu%2Fdata%2Fset%2F87c14712-9239-4176-a112-913ff797481b%22%7D%7D%5D%7D&page=2&pageLength=20&sort=archiveSortId",
        "type": "OrderedCollectionPage"
    }
}
```

**Comment by kamerynB** [comment]
@roamye @jffcamp This requires a hotfix per [this bugherd](https://www.bugherd.com/projects/284041/tasks/2917). I will be working on it now. I believe this can be fixed via the MT only.

**Comment by kamerynB** [comment]
@roamye @jffcamp Hotfix has been deployed.

**Comment by jffcamp** [comment]
@kamerynB, is it a problem that Litchfield is at the bottom of the list?

![Image](https://github.com/user-attachments/assets/8cc7135a-352e-43ae-894a-105e0773cb9b)

**Comment by kamerynB** [comment]
@jffcamp I don't think so. I think that is expected based on the fact that ML is now sorting the results with sets and objects together, rather than separately and then having the frontend combine the results.

**Comment by clarkepeterf** [comment]
@jffcamp While the rest of the results look alphabetical, they are actually being sorted based on a sort identifier in the data. Litchfield has that largest identifier so it is last. If it should be sorted differently, the data needs to change.

Here are the sort identifiers from the data:

- Bridgeport
  - 00000fa0000013880000<ins>0</ins>3e8
- Westville
  - 00000fa0000013880000<ins>1</ins>f40
- Litchfield
  - 00000fa0000013880000<ins>2</ins>328

You can see that the Litchfield has a larger identifier so it comes later in the list. The identifiers are the same until the 4th to last character which I have underlined.

**Comment by jffcamp** [comment]
[like]  Campbell, Jeffrey reacted to your message:
________________________________

**Comment by clarkepeterf** [comment]
So, possibly a data issue for the library? I would imagine they are meant to be alphabetical.

**Comment by clarkepeterf** [comment]
@kkdavis14  Based on [my previous comment](https://github.com/project-lux/lux-frontend/issues/458#issuecomment-2755498264) do you think this would be a data issue for the library?

**Comment by kkdavis14** [comment]
@clarkepeterf they are not meant to be alphabetical--they should be sorted based on the sort identifier. glad the combining of scopes is happening! thanks!

**Comment by clarkepeterf** [comment]
I know, I thought perhaps the library put in the wrong identifier for Litchfield

**Comment by kkdavis14** [comment]
@clarkepeterf @jffcamp Archives at Yale view matches PRD, so that's what we want to shoot for. 

<img width="344" alt="Image" src="https://github.com/user-attachments/assets/f220b6d4-e652-4ceb-b4f3-592a6f1342da" />

**Comment by clarkepeterf** [comment]
Summary is - yes we match the Yale Archive view. Perhaps Litchfield is wrong. But they have other things to deal with and this is low priority so we won't send anything to them.

**Comment by roamye** [comment]
Approved by UAT

**Comment by roamye** [comment]
Looks good in PRD. Closing.
